src/main.rs

```diff
@@ -127,6 +133,56 @@ struct CameraErr {
     err: neolink::Error,
 }
 
+fn set_up_tls(config: &Config, rtsp: &RtspServer) {
+    let tls_client_auth = match &config.tls_client_auth as &str {
+        "request" => TlsAuthenticationMode::Requested,
+        "require" => TlsAuthenticationMode::Required,
+        "none" => TlsAuthenticationMode::None,
+        _ => unreachable!(),
+    };
+    if let Some(cert_path) = &config.certificate {
+        rtsp.set_tls(&cert_path, tls_client_auth).expect("Failed to set up TLS");
+    }
+}
+
+fn set_up_users(users: &Vec<UserConfig>, rtsp: &RtspServer) {
+    // Setting up users
+    let mut credentials = vec![];
+    for user in users {
+        let name = &user.name;
+        let pass = &user.pass;
+        let user_pass = match (name, pass)  {
+            (Some(name), Some(pass)) => Some((&name as &str, &pass as &str)),
+            _ => None,
```

Comment:
Might be worth a `warn!()` here if you get `(Some(), None)` or `(None, Some())` to let the user know they dun goofed.

Reply:
Rule of thumb: every warning, every sanity check, will save you from dealing with at least one trouble ticket.

Reply:
That means a warn and a return None. Would that need an enclosure I wonder or something else. More rust foo to learn in the docs :).

Reply:
Yeah, you would do:
```rust
(Some(_), None) | (None, Some(_)) => {
    warn!("Username and password must be supplied together - ignoring [[users]] entry");
    None
}
```

because the block evaluates to the last statement in it.

Reply:
I have added this in but it should also be caught by the validation